packages/web/app/src/lib/preflight-sandbox/preflight-script-worker.ts

```diff
@@ -116,17 +125,44 @@ async function execute(args: WorkerEvents.Incoming.EventData): Promise<void> {
     },
     environment: {
       get(key: string) {
-        return Object.freeze(workingEnvironmentVariables[key]);
+        return Object.freeze(workerData.environmentVariables[key]);
       },
       set(key: string, value: unknown) {
         const validValue = getValidEnvVariable(value);
         if (validValue === undefined) {
-          delete workingEnvironmentVariables[key];
+          delete workerData.environmentVariables[key];
         } else {
-          workingEnvironmentVariables[key] = validValue;
+          workerData.environmentVariables[key] = validValue;
         }
       },
     },
+    /**
+     * Helpers for manipulating the request before it is sent.
+     */
+    request: {
+      /**
+       * Helpers for manipulating the request headers.
+       */
+      headers: {
+        /**
+         * Add one header to the request.
+         *
+         * @param header - The name of the header.
+         * @param value - The value of the header.
+         */
+        add: (header: string, value: string) => {
+          workerData.headers.append(header, value);
+        },
+        /**
+         * Add multiple headers to the request.
+         *
+         * @param headersInit - The {@link HeadersInit} to add.
+         */
+        addMany: (headersInit: HeadersInit) => {
```

Comment:
Can't we just let people access `Headers` that they are familiar with? With your approach we need to add methods like `delete`/`remove`, `get` and `has` and they need to learn new API.

Reply:
We should probably give them access to a [Headers object](https://developer.mozilla.org/en-US/docs/Web/API/Headers/Headers). 

We can then serialize it to a JSON object and forward it to the main thread.

Reply:
Yes, entries -> Headers -> entries

Reply:
Yes we can give them access to data to mutate.

Given this [comment](https://github.com/graphql-hive/console/pull/6378#discussion_r1920285682) I am no longer sure it should be `Headers` however.

Based on the resolution there, we can expose a reference of type `Record<string, string>` or `Headers`. TBD.

Reply:
Give this feedback on headers, why then were environment variables treated differently?

https://github.com/graphql-hive/console/blob/main/packages/web/app/src/lib/preflight-sandbox/preflight-script-worker.ts#L117-L129

Reply:
We wanted to merge Preflight Script asap and it was very chaotic process

Reply:
Now we want to incrementally add new features and improve logic there